python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -223,11 +223,12 @@ def get_asset_nodes_by_asset_key(
             external_asset_node.asset_key, (None, None, None)
         )
         if preexisting_asset_node is None or preexisting_asset_node.is_source:
-            asset_nodes_by_asset_key[external_asset_node.asset_key] = (
-                repo_loc,
-                repo,
-                external_asset_node,
-            )
+            if asset_keys is not None and external_asset_node.asset_key in asset_keys:
+                asset_nodes_by_asset_key[external_asset_node.asset_key] = (
+                    repo_loc,
+                    repo,
+                    external_asset_node,
+                )
 
     asset_checks_loader = AssetChecksLoader(
```

Comment:
filtering this down is the key

Reply:
I've updated the graphql server setup so we can use async resolvers, which sets us up to for an async/await backed "DataLoader" pattern where you accumulate all the concurrent resolver calls across a sequence of objects and then do one dispatch to fetch the set 
https://docs.graphene-python.org/en/latest/execution/dataloader/